File: src/main/java/io/github/takenoha/towerdefense/paper/TeamManagementConfirmationHolder.java
package io.github.takenoha.towerdefense.paper;

import java.util.Objects;
import java.util.UUID;
import org.bukkit.inventory.Inventory;
import org.bukkit.inventory.InventoryHolder;

/** Carries the target and action for a team-management confirmation screen. */
public final class TeamManagementConfirmationHolder implements InventoryHolder {
    public enum Action {
        REMOVE_MEMBER,
        TRANSFER_OWNER,
        LEAVE_TEAM
    }

    private final UUID coreId;
    private final UUID targetId;
    private final Action action;
    private Inventory inventory;

    public TeamManagementConfirmationHolder(UUID coreId, UUID targetId, Action action) {
        this.coreId = Objects.requireNonNull(coreId, "coreId");
        this.targetId = Objects.requireNonNull(targetId, "targetId");
        this.action = Objects.requireNonNull(action, "action");
    }

    public UUID coreId() {
        return coreId;
    }

    public UUID targetId() {
        return targetId;
    }

    public Action action() {
        return action;
    }

    void attach(Inventory inventory) {
        this.inventory = Objects.requireNonNull(inventory, "inventory");
    }

    @Override
    public Inventory getInventory() {
        if (inventory == null) {
            throw new IllegalStateException("the GUI inventory has not been attached");
        }
        return inventory;
    }
}

File: src/main/kotlin/io/github/takenoha/towerdefense/paper/TeamManagementConfirmationHolder.kt
package io.github.takenoha.towerdefense.paper

import java.util.Objects
import java.util.UUID
import org.bukkit.inventory.Inventory
import org.bukkit.inventory.InventoryHolder

/** Carries the target and action for a team-management confirmation screen. */
class TeamManagementConfirmationHolder(
    coreId: UUID,
    targetId: UUID,
    action: Action,
) : InventoryHolder {
    enum class Action {
        REMOVE_MEMBER,
        TRANSFER_OWNER,
        LEAVE_TEAM,
    }

    private val coreIdValue: UUID = Objects.requireNonNull(coreId, "coreId")
    private val targetIdValue: UUID = Objects.requireNonNull(targetId, "targetId")
    private val actionValue: Action = Objects.requireNonNull(action, "action")
    private var inventory: Inventory? = null

    fun coreId(): UUID = coreIdValue

    fun targetId(): UUID = targetIdValue

    fun action(): Action = actionValue

    fun attach(inventory: Inventory) {
        this.inventory = Objects.requireNonNull(inventory, "inventory")
    }

    override fun getInventory(): Inventory =
        inventory ?: throw IllegalStateException("the GUI inventory has not been attached")
}
